civiclens/tests/test_upload_bulk_csvs_to_db.py

```diff
@@ -0,0 +1,217 @@
+import pytest
+from unittest.mock import patch
+from civiclens.data_engineering import upload_bulk_csvs_to_db
+
+
+def test_get_document_objectId_fakeid(mocker):
+    fake_id = "NOT_REAL"
+
+    mock_response = []
+    with patch(
+        "civiclens.data_engineering.access_api_data.pull_reg_gov_data"
+    ) as mock_api_call:
+        mock_api_call.return_value = mock_response
+
+        # mocked API call
+        with pytest.raises(IndexError) as e:
+            upload_bulk_csvs_to_db.get_document_objectId(fake_id)
+
+        assert str(e.value) == "list index out of range"
+        # we get this error because the API response returns an empty list
+
+
+def test_get_document_objectId_multiple_ids(mocker):
+
+    mock_response = []
+    with patch(
+        "civiclens.data_engineering.access_api_data.pull_reg_gov_data"
+    ) as mock_api_call:
+        mock_api_call.return_value = mock_response
+
+        # mocked API call
+        with pytest.raises(IndexError) as e:
+            upload_bulk_csvs_to_db.get_document_objectId(
+                ("FDA-2017-V-5183-0001", "GSA-GSA-2019-0002-0028")
+            )
+
+        assert str(e.value) == "list index out of range"
+        # we get this error because the API response returns an empty list
+
+
+def test_get_document_objectId_working_id(mocker):
+    # Define the input ID and expected output
+    input_id = "FDA-2017-V-5183-0001"
+    expected_output = "0900006482ab30a2"
+
+    mock_api_call_response = "0900006482ab30a2"
+    with patch(
+        "civiclens.data_engineering.upload_bulk_csvs_to_db.get_document_objectId"
+    ) as mock_api_call:
+        mock_api_call.return_value = mock_api_call_response
+
+        # Call the function being tested
+        result = upload_bulk_csvs_to_db.get_document_objectId(input_id)
+
+        # Assert that the function returned the expected output
+        assert result == expected_output
+
+        # Assert that the API call function was called with the correct arguments
+        mock_api_call.assert_called_once_with(input_id)
+
+
+def test_format_date_no_date():
+    try:
+        upload_bulk_csvs_to_db.format_date("")
+    except Exception as e:
+        # print(e)
+        assert str(e) == "Invalid isoformat string: ''"
+
+
+def test_format_date_good_date():
+    assert (
+        upload_bulk_csvs_to_db.format_date("2024-03-06T05:00:00")
+        == "2024-03-06T05:00:00Z"
+    )
+
+
+def test_format_date_bad_date():
+    try:
+        upload_bulk_csvs_to_db.format_date("2024--06T05:00:00")
+    except Exception as e:
+        # print(e)
+        assert str(e) == "Invalid isoformat string: '2024--06T05:00:00'"
+
+
+def test_extract_fields_from_row_no_data():
+    try:
+        upload_bulk_csvs_to_db.extract_fields_from_row("", "test_id")
+    except Exception as e:
+        assert str(e) == "string indices must be integers, not 'str'"
+
+
+def test_extract_fields_from_row_good_data():
+    pl_row = {
```

Comment:
From a readability perspective, it'd be better to save this in a file as opposed to directly putting it in the python 🐍 file. Add at data folder to your `tests` folder and store it there.